fl4health/utils/nnunet_utils.py

```diff
@@ -0,0 +1,364 @@
+    See https://github.com/MIC-DKFZ/nnUNet/tree/v2.5.1
+    """
+
+    _2D = "2d"
+    _3D_FULLRES = "3d_fullres"
+    _3D_CASCADE = "3d_cascade_fullres"
+    _3D_LOWRES = "3d_lowres"
+
+
+NNUNET_DEFAULT_NP = {  # Nnunet's default number of processes for each config
+    NnunetConfig._2D: 8,
+    NnunetConfig._3D_FULLRES: 4,
+    NnunetConfig._3D_CASCADE: 4,
+    NnunetConfig._3D_LOWRES: 8,
+}
+
+NNUNET_N_SPATIAL_DIMS = {  # The number of spatial dims for each config
+    NnunetConfig._2D: 2,
+    NnunetConfig._3D_FULLRES: 3,
+    NnunetConfig._3D_CASCADE: 3,
+    NnunetConfig._3D_LOWRES: 3,
+}
+
+
+def use_default_signal_handlers(fn: Callable) -> Callable:
+    """
+    This is a decorator that resets the SIGINT and SIGTERM signal handlers back to the
+    python defaults for the execution of the method
+
+    flwr 1.9.0 overrides the default signal handlers with handlers that raise an error
+    on any interruption or termination. Since nnunet spawns child processes which
+    inherit these handlers, when those subprocesses are terminated (which is expected
+    behaviour), the flwr signal handlers raise an error (which we don't want).
+
+    Flwr is expected to fix this in the next release. See the following issue:
+    https://github.com/adap/flower/issues/3837
+    """
+
+    def new_fn(*args: Any, **kwargs: Any) -> Any:
+        # Set SIGINT and SIGTERM back to defaults. Method returns previous handler
+        sigint_old = signal.signal(signal.SIGINT, signal.default_int_handler)
+        sigterm_old = signal.signal(signal.SIGTERM, signal.SIG_DFL)
+        # Execute function
+        output = fn(*args, **kwargs)
+        # Reset handlers back to what they were before function call
+        signal.signal(signal.SIGINT, sigint_old)
+        signal.signal(signal.SIGTERM, sigterm_old)
+        return output
+
+    return new_fn
+
+
+def reload_modules(packages: Sequence[str]) -> None:
+    """
+    Given the names of one or more packages, subpackages or modules, reloads all the
+    modules within the scope of each package or the modules themselves if a module was
+    specified.
+
+    Args:
+        package (Sequence[str]): The absolute names of the packages, subpackages or
+            modules to reload. The entire import hierarchy must be specified. Eg.
+            'package.subpackage' to reload all modules in subpackage, not just
+            'subpackage'. Packages are reloaded in the order they are given
+    """
+    for m_name, module in list(sys.modules.items()):
+        for package in packages:
+            if m_name.startswith(package):
+                try:
+                    reload(module)
+                except Exception as e:
+                    log(DEBUG, f"Failed to reload module {m_name}: {e}")
+
+
+def set_nnunet_env(verbose: bool = False, **kwargs: str) -> None:
+    """
+    For each keyword argument name and value sets the current environment variable with
```

Comment:
Maybe just to make it really explicit why this is happening you can mention that env variables need to be set prior to importing (or reloading)